controllers/humiocluster_controller.go

```diff
@@ -447,31 +448,48 @@ func (r *HumioClusterReconciler) validateNodeCount(hc *humiov1alpha1.HumioCluste
 func (r *HumioClusterReconciler) ensureExtraKafkaConfigsConfigMap(ctx context.Context, hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool) error {
 	extraKafkaConfigsConfigMapData := hnp.GetExtraKafkaConfigs()
 	if extraKafkaConfigsConfigMapData == "" {
+		extraKafkaConfigsConfigMap, err := kubernetes.GetConfigMap(ctx, r, hnp.GetExtraKafkaConfigsConfigMapName(), hc.Namespace)
+		if err == nil {
+			// TODO: refactor and move deletion to cleanupUnusedResources
```

Comment:
Is this TODO something it should be done now, or is it a future improvement?

Reply:
My intention is to keep the code as it is in the current state of the PR/branch, mostly because the rolePermissions and viewGroupPermissions code follow the same style in the code. I've also updated the sections for rolePermissions and viewGroupPermissions to include the same TODO comment. The idea is that we can at a later stage when we refactor one of them, we can refactor all. The diff was just much smaller to "do the same" for extraKafkaConfigs than having to redo rolePermissions and viewGroupPermissions as well.